### PR TITLE
[SPARK-22036][SQL][FOLLOWUP] Fix decimalArithmeticOperations.sql

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Fix decimalArithmeticOperations.sql test

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by wangyum** [comment]
retest please.

**Comment by wangyum** [comment]
retest this please.

File: sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql [hunk 1]
```diff
@@ -49,7 +49,7 @@ select 1e35 / 0.1;
 
 -- arithmetic operations causing a precision loss are truncated
 select 123456789123456789.1234567890 * 1.123456789123456789;
-select 0.001 / 9876543210987654321098765432109876543.2
+select 0.001 / 9876543210987654321098765432109876543.2;
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql]: Hi, @wangyum .
If this is only for a specific test file, can we have more proper title? `Fix decimalArithmeticOperations.sql`?

Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql]: ... A good catch! We need to review the PR more carefully.

Comment (gatorsmile) [reply sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql]: This is not a good test case. The results and schemas are the same no matter whether we set  `spark.sql.decimalOperations.allowPrecisionLoss` to true or false.

**Comment by mgaido91** [comment]
@wangyum yes, you're right. I don't know how I missed it, I am very sorry for my mistake. Anyway, this is fixed also in this PR #20350, which is the "second part" of making Spark SQL-2011 compliant for decimal operations.

**Comment by gatorsmile** [comment]
@mgaido91 We just remove this test case? I think we can still keep it, right?

**Comment by mgaido91** [comment]
@gatorsmile because this is an example of overflow, ie. what is covered in the new PR: in the new PR I added many tests for this case, so I felt this unnecessary.

**Comment by gatorsmile** [comment]
@mgaido91 If we remove it, we still need test cases for verifying the effects of `spark.sql.decimalOperations.allowPrecisionLoss`.

**Comment by mgaido91** [comment]
@gatorsmile yes, I see what you mean, we should keep it to check that changing `spark.sql.decimalOperations.allowPrecisionLoss` doesn't affect this case, you are right. I will comment on my PR and update it accordingly if you agree, thanks.

**Comment by gatorsmile** [comment]
If possible, I prefer to doing it in this PR. We should merge this test-only PR to 2.3 and master for verifying the behavior of the changes made in 2.3.

**Comment by mgaido91** [comment]
Yes sure,I'll rebase my PR after this is merged.

File: sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql [hunk 2]
```diff
@@ -48,8 +48,9 @@ select 12345678901234567890.0 * 12345678901234567890.0;
 select 1e35 / 0.1;
 
 -- arithmetic operations causing a precision loss are truncated
+select 12345678912345678912345678912.1234567 + 9999999999999999999999999999999.12345;
```
Comment (wangyum) [inline sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql]: The result is:
```
-- !query 17
select 12345678912345678912345678912.1234567 + 9999999999999999999999999999999.12345
-- !query 17 schema
struct<(CAST(12345678912345678912345678912.1234567 AS DECIMAL(38,6)) + CAST(9999999999999999999999999999999.12345 AS DECIMAL(38,6))):decimal(38,6)>
-- !query 17 output
10012345678912345678912345678911.246907


-- !query 18
select 123456789123456789.1234567890 * 1.123456789123456789
-- !query 18 schema
struct<(CAST(123456789123456789.1234567890 AS DECIMAL(36,18)) * CAST(1.123456789123456789 AS DECIMAL(36,18))):decimal(38,18)>
-- !query 18 output
138698367904130467.654320988515622621


-- !query 19
select 12345678912345.123456789123 / 0.000000012345678
-- !query 19 schema
struct<(CAST(12345678912345.123456789123 AS DECIMAL(29,15)) / CAST(1.2345678E-8 AS DECIMAL(29,15))):decimal(38,9)>
-- !query 19 output
1000000073899961059796.725866332
```

File: sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql [hunk 3]
```diff
@@ -74,7 +75,8 @@ select 12345678901234567890.0 * 12345678901234567890.0;
 select 1e35 / 0.1;
 
 -- arithmetic operations causing a precision loss return NULL
+select 12345678912345678912345678912.1234567 + 9999999999999999999999999999999.12345;
```
Comment (wangyum) [inline sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/decimalArithmeticOperations.sql]: The result is:
```
-- !query 32
select 12345678912345678912345678912.1234567 + 9999999999999999999999999999999.12345
-- !query 32 schema
struct<(CAST(12345678912345678912345678912.1234567 AS DECIMAL(38,7)) + CAST(9999999999999999999999999999999.12345 AS DECIMAL(38,7))):decimal(38,7)>
-- !query 32 output
NULL


-- !query 33
select 123456789123456789.1234567890 * 1.123456789123456789
-- !query 33 schema
struct<(CAST(123456789123456789.1234567890 AS DECIMAL(36,18)) * CAST(1.123456789123456789 AS DECIMAL(36,18))):decimal(38,28)>
-- !query 33 output
NULL


-- !query 34
select 12345678912345.123456789123 / 0.000000012345678
-- !query 34 schema
struct<(CAST(12345678912345.123456789123 AS DECIMAL(29,15)) / CAST(1.2345678E-8 AS DECIMAL(29,15))):decimal(38,18)>
-- !query 34 output
NULL
```

**Comment by mgaido91** [comment]
LGTM

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master/2.3!

**Pull request closed without merging**